java/src/jmri/jmrit/logixng/util/parser/functions/ConvertFunctions.java

```diff
@@ -163,4 +164,43 @@ public String getDescription() {
         
     }
     
+    public static class Hex2DecFunction implements Function {
+        
+        @Override
+        public String getModule() {
+            return new ConvertFunctions().getModule();
+        }
+        
+        @Override
+        public String getConstantDescriptions() {
+            return new ConvertFunctions().getConstantDescription();
+        }
+        
+        @Override
+        public String getName() {
+            return "hex2dec";
+        }
+        
+        @Override
+        public Object calculate(SymbolTable symbolTable, List<ExpressionNode> parameterList)
+                throws JmriException {
+            
+            if (parameterList.size() != 1) {
+                throw new WrongNumberOfParametersException(Bundle.getMessage("WrongNumberOfParameters2", getName(), 1));
+            }
+            Object o = parameterList.get(0).calculate(symbolTable);
+            if (o != null) {
+                return Long.parseLong(o.toString(), 16);
```

Comment:
This can throw a NumberFormatException if the input isn't parseable.  Does that need to be handled here?

Reply:
Thanks for asking.

There are a lot of things that can go wrong with LogixNG. Therefore LogixNG has a solid error handling:
https://www.jmri.org/help/en/html/tools/logixng/ErrorHandling.shtml

In this case, it could be useful to give the user the option to handle the error, for example by adding the parameters:
`hex2dec(value, resultIfNull, resultIfError)`

where `resultIfNull` is the result to be returned is `value` is null and `resultIfError` is the result to be returned if `value` isn't parsable. But these two extra parameters can be added later. See the `sin()` function that can take 1 to 4 parameters:
https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/util/parser/functions/FunctionsBundle.properties#L87

But before I add this, I need to think it thru to get it right. So this PR is ready to merge.